crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -964,19 +964,22 @@ fn compute_limited(
 
         let remote_events = &remote_room.timeline;
         if remote_events.is_empty() {
-            trace!("no timeline updates in the response => not limited");
+            trace!(
+                %room_id,
+                "no timeline updates in the response => not limited"
+            );
```

Comment:
Instead of adding it to all the individual `trace!` calls, why not add it as a field?
`#[instrument(skip_all, fields(room_id))]` + `Span::current().record("room_id", tracing::field::debug(room_id));` (note also: we use debug, not display, for room / user / ... IDs)

Reply:
I definitely prefer it with the record, but if you want we can keep it like this (with `?` instead of `%`).